lib/integrations/drivers/github.rb

```diff
@@ -0,0 +1,58 @@
+### Github integration
+module Integrations::Drivers
+  class Github < Integrations::BaseDriver
+    def self.display_name
+      'Github'
+    end
+
+    def type
+      :github
+    end
+
+    def api_url
+      'https://api.github.com/'
+    end
+
+    def create_task(title)
+      configuration = self.configuration
+      url =  api_url + 'repos/' + configuration["username"].to_s + '/' + configuration["selected_application"].to_s + '/issues'
+      payload = { title: title }.to_json
+      headers = { "Authorization": "token " + configuration["token"].to_s, "Content-Type": "application/json" }
+      response = RestClient.post url, payload, headers rescue nil
+      if response
+        response = JSON.parse(response)
+        response
+      end
+    end
+
+    def applications
+      configuration = self.configuration
+      url = api_url + 'user/repos?affiliation=owner,organization_member'
+      headers = { "Authorization": "token " + configuration["token"].to_s }
+      response = RestClient.get url, headers rescue nil
```

Comment:
You should make 2 methods get and post and set the headers there That way you can just send the url and the data if needed.